platform/darwin/src/darwin_conversion.hpp

```diff
@@ -0,0 +1,132 @@
+#pragma once
+
+#include <mbgl/util/logging.hpp>
+#include <mbgl/style/conversion.hpp>
+#include <mbgl/util/feature.hpp>
+#include <mbgl/util/optional.hpp>
+
+namespace mbgl {
+namespace style {
+namespace conversion {
+
+// A minimal wrapper class conforming to the requirements for `objectMember(v, name)` (see mbgl/style/conversion.hpp)
+// This is necessary because using `NSObject*` as the value type in `optional<NSObject*>` causes problems for the ARC,
+// due to things like `optional(const value_type& __v)`
+class OptionalNSObjectValue {
+public:
+    OptionalNSObjectValue(NSObject *_value) : value(_value) {}
```

Comment:
Annotate `value`’s type as `_Nullable` and surround the header’s contents in `NS_ASSUME_NONNULL_BEGIN` and `NS_ASSUME_NONNULL_END` to enable better compiler/analyzer diagnostics.